malariagen_data/anoph/fst.py

```diff
@@ -123,13 +124,62 @@ def fst_gwss(
         inline_array: base_params.inline_array = base_params.inline_array_default,
         chunks: base_params.chunks = base_params.native_chunks,
         clip_min: fst_params.clip_min = 0.0,
+        contig: Optional[base_params.region] = None,  # Deprecated
     ) -> Tuple[np.ndarray, np.ndarray]:
         # Change this name if you ever change the behaviour of this function, to
         # invalidate any previously cached data.
-        name = "fst_gwss_v2"
+        name = "fst_gwss_v3"
+
+        # Get a copy of the local variables, which will include all provided function parameters.
+        local_vars = locals().copy()
+
+        # Specify which quasi-positional args are required.
+        required_args = ("window_size", "cohort1_query", "cohort2_query")
+
+        # Raise an error for any missing required args.
+        missing_args = []
+        for required_arg in required_args:
+            if local_vars.get(required_arg) is None:
+                missing_args.append(required_arg)
+        if missing_args:
+            raise ValueError(f"Missing required arguments: {missing_args}")
```

Comment:
This shouldn't be necessary I don't think, if the type annotations are left the same.